### DISCUSSION TITLE
Workaround for naming collisions?

### DISCUSSION THREAD

**Discussion by joeflatt** [timostamm/protobuf-ts#160] [Q&A] [answered]
Hi,

I'm new to this so not sure if it's a bug or a known issue that has a workaround.  

I compile against a large set of proto files that contain a message named Object.  I don't have any control over the proto files themselves.  The generated typescript code won't compile because it makes use of static methods on JavaScript's Object class (i.e. Object.defineProperty).

Here's how I'm using protobuf-ts:
npx protoc --ts_out src/ --ts_opt generate_dependencies --proto_path apis/ `find apis -name '*.proto'`

Thanks!

**Comment by timostamm** [comment, marked as answer]
That's a bug. Thanks for the report! Tracking in #161.

As a workaround, you should be fine with `--ts_opt force_optimize_speed`. If I'm not mistaken, this doesn't generate `Object.defineProperty`.

**Reply by joeflatt** [reply]
Thanks for the quick response!  force_optimize_speed still includes Object.defineProperty... but I'm able to move ahead by not including the Object message for the time being.

**Reply by timostamm** [reply]
Ah, sorry, it's `force_optimize_code_size`, not `force_optimize_speed`. 

No need for that workaround though, just released v2.0.6 with a fix.
